src/coreclr/jit/gentree.cpp

```diff
@@ -15683,6 +15683,10 @@ GenTree* Compiler::gtNewTempAssign(
         {
             ok = true;
         }
+        else if ((dstTyp == TYP_I_IMPL) && varTypeIsGC(valTyp))
```

Comment:
> `varTypeIsGC`

This includes `TYP_REF` as well, do we need that to fix the asserts?

Reply:
I guess not? I can just do it by BYREF only.